_source/_posts/2022-08-08-react-redux.md

```diff
@@ -0,0 +1,528 @@
+```bash
+npm add -D @types/react-redux@7.1
+```
+
+4. We'll also need routing (and routing types) for our app:
+```bash
+npm install react-router-dom@5
+npm add -D @types/react-router-dom@5.3
+```
+
+5. To start your app, run:
+```bash
+npm start
+```
+
+## How Redux works
+
+To update state using Redux, an action is dispatched and the store then uses the root reducer to calculate a new state as compared to the old state, which then notifies the proper subscribers of the update so that the UI can properly be updated.
+
+After the UI for your application is initially rendered using an initial global Redux store created using the root reducer, a typical redux workflow goes like this:
+
+1. An event in the UI such as a user button interaction happens in your app which triggers an action.
+2. The action is then dispatched to the store and an event handler that handles the needed logic makes the changes to your global state.
+3. The store notifies subscribed parts in the UI that there has been an update.
+4. Those subscribed parts then check to see if a re-render is needed to update whatever needs updating in the UI.
+
+**NOTE:** The built-in Redux template we used includes a great counter example of how Redux works. You can take a closer look at [Redux's explanation of the counter example](https://redux.js.org/introduction/examples/).
+
+### Useful Terms
+
+**[Action](https://redux.js.org/tutorials/essentials/part-1-overview-concepts#reducers)**
+
+An Action is an JS object with a type and a payload. It describes an actionable interaction within your application, and typically is named something like `createUser` or `addToDo`.
+
+**[Action Creators](https://redux.js.org/tutorials/essentials/part-1-overview-concepts#reducers)**
+
+Action Creators can be used to dynamically create actions. NOT IN APP
+
+**[Reducers](https://redux.js.org/tutorials/essentials/part-1-overview-concepts#reducers)**
+
+A Reducer returns the proper current state with applicable changes as compared to the previous state when it's passed a new state. They work similarly to the Array.reduce method.
+
+**[Redux Store](https://redux.js.org/tutorials/fundamentals/part-4-store#redux-store)**
+
+The current live state for your app lives in the Redux Store.
+
+**[Selectors](https://redux.js.org/tutorials/essentials/part-1-overview-concepts#selectors)**
+
+A selector returns a piece of the live Redux state as stored in the live Redux store.
+
+**[Dispatch](https://redux.js.org/tutorials/essentials/part-1-overview-concepts#dispatch)**
+
+A Dispatch is a method with triggers an action that in turn updates the Redux store.
+
+## Add authentication using OAuth2 and OIDC
+
+For this demo app, we'll be using [Okta's SPA redirect model](https://developer.okta.com/docs/guides/sign-into-spa-redirect/react/main/) to authenticate and fetch user info.
+
+{% include setup/cli.md type="spa" %}
+
+## Application Setup with State Management
+
+## Setup Environment Variables
```

Comment:
Do you think this header is needed? If so, we can think of some other header titles, but if you don't feel strongly, I might suggest removing since the keywords won't be valuable here

Reply:
Don't think it's needed, remove it.